testgrid/cmd/updater/main.go

```diff
@@ -266,30 +315,46 @@ func AppendColumn(headers []string, grid *state.Grid, rows map[string]*state.Row
 		missing[name] = row
 	}
 
-	for name, result := range build.Results {
-		delete(missing, name)
-		r, ok := rows[name]
-		if !ok {
-			r = &state.Row{
-				Name: name,
-				Id:   name,
+	found := map[string]bool{}
+
+	for target, results := range build.Rows {
+		for _, br := range results {
+			prefix := br.Format(format, build.Metadata)
+			name := prefix
+			var idx int
+			for {
```

Comment:
`for idx := 1; !found[name]; idx++ { name = fmt.Sprintf("%s [%d]", prefix, idx }` ?

Reply:
This should really have a comment.

Reply:
Done